lib/chai/core/assertions.js

```diff
@@ -529,16 +529,35 @@ module.exports = function (chai, _) {
   function assertAbove (n, msg) {
     if (msg) flag(this, 'message', msg);
     var obj = flag(this, 'object')
-      , doLength = flag(this, 'doLength');
+      , doLength = flag(this, 'doLength')
+      , objType = _.type(obj)
+      , nType = _.type(n)
+      , msgPrefix = (msg) ? msg + ': ' : '';
 
     if (doLength) {
       new Assertion(obj, msg).to.have.property('length');
     } else {
-      new Assertion(obj, msg).is.a('number');
+      // Since utils/expectTypes is missing from 4.x.x and no better way to assert multiple types
+      if (!(objType === 'number' || objType === 'date')) {
+        throw new Error(msgPrefix + 'expected ' + obj + ' to be a number or a date');
+      }
     }
 
-    if (_.type(n) !== 'number') {
-      throw new Error('the argument to above must be a number');
+    if (!(nType === 'number' || nType === 'date')) {
+      var shoudBeType = (objType === 'number') 
+        ? 'number'
+        : (objType === 'date') 
+          ? 'date' 
+          : 'number or date';
```

Comment:
Sure, no problem :) I know that there are plans to introduce a linter, would definitely be beneficial for things such cases, no more "nits" and stuff.
Also, you guys can take a look at [prettier](https://github.com/prettier/prettier) as an alternative to standard linter.